test/e2e/services.go

```diff
@@ -41,3 +46,42 @@ func newRulesObjstoreService(e e2e.Environment) (*e2e.InstrumentedRunnable, erro
 		},
 	), nil
 }
+
+func newMinio(t *testing.T, e e2e.Environment) {
+	// Create S3 replacement for rules backend
+	bucket := "obs_rules_test"
+	userID := strconv.Itoa(os.Getuid())
+	ports := map[string]int{e2edb.AccessPortName: 8090}
+	envVars := []string{
+		"MINIO_ROOT_USER=" + e2edb.MinioAccessKey,
+		"MINIO_ROOT_PASSWORD=" + e2edb.MinioSecretKey,
+		"MINIO_BROWSER=" + "off",
+		"ENABLE_HTTPS=" + "0",
+		// https://docs.min.io/docs/minio-kms-quickstart-guide.html
+		"MINIO_KMS_KES_ENDPOINT=" + "https://play.min.io:7373",
```

Comment:
All of this is very fancy, but I wonder why we need Minio encryption for this feature of adding metrics? It is kind of orthogonal to this project. I'm essentially worried about adding an extra dependency on another free service to our e2e tests. Essentially, if https://pllay.mini.io:7373 is ever down, then our e2e tests will fail. If testing encryption doesn't add any benefit / confidence to the project (since it's orthogonal) then I would prefer to remove the complexity / extra dependency.

Reply:
Hmm. I see your point!

I investigated this a bit, as to why we're using KMS here! So we use https://github.com/efficientgo/e2e for our e2e tests, which has a [`NewMinio`](https://github.com/efficientgo/e2e/blob/main/db/db.go#L45) method, which uses KMS internally too. This module was initially [ported from cortex](https://github.com/cortexproject/cortex/blob/f639b1855c9f0c9564113709a6bce2996d151ec7/integration/e2e/db/db.go), where KMS was first added to support AWS S3 server-side encryption as per this PR: https://github.com/cortexproject/cortex/pull/3651.

There were issues earlier with it in `efficientgo/e2e` which were fixed in `observatorium/api` which I was copied over here, but seems like it was fixed in `e2e` repo too, so I can just bump the version here: https://github.com/efficientgo/e2e/pull/26.

But yes, this should be configurable, as KMS isn't needed for cases like this! I can raise a PR to `efficientgo/e2e`, and update the version across e2e tests in diff repos in `Observatorium`, once that gets merged. :)

Reply:
Updated this PR to use latest version of `e2e` and not have the special minio method, but used the fixed one from there: https://github.com/observatorium/rules-objstore/pull/9/commits/3a92bb1f99f43a55ce2117ab04d87f8e25a74908

Reply:
Thanks for the thorough work @saswatamcode